light/client.go

```diff
@@ -1066,13 +1098,13 @@ func (c *Client) compareFirstHeaderWithWitnesses(ctx context.Context, h *types.S
 		case nil:
 			continue
 		case errConflictingHeaders:
-			c.logger.Error(fmt.Sprintf(`witness #%d has a different header. Please check primary is correct
-and remove witness. Otherwise, use a different primary`, e.WitnessIndex), "witness", c.witnesses[e.WitnessIndex])
+			c.logger.Error(fmt.Sprintf(`witness #%d has a different header (%X). Please check primary is correct
```

Comment:
we should avoid `Sprintf` here and just pass key vals. e.g. `Error("witness has different header...", "witness", ..., "header", ...)`.

Reply:
ok 👍